extensions-contrib/kubernetes-overlord-extensions/src/test/java/org/apache/druid/k8s/overlord/KubernetesTaskRunnerTest.java

```diff
@@ -599,4 +605,26 @@ public TaskLocation getLocation()
 
     verifyAll();
   }
+
+  @Test
+  public void test_metricsReported_whenTaskStateChange()
+  {
+    KubernetesWorkItem workItem = new KubernetesWorkItem(task, null) {
+      @Override
+      public TaskLocation getLocation()
+      {
+        return TaskLocation.unknown();
+      }
+    };
+
+    runner.tasks.put(task.getId(), workItem);
+
+    emitter.emit(EasyMock.anyObject(ServiceEventBuilder.class));
```

Comment:
Here we basically measure the `emitter` has been called with whatever param.
We don't verify the exact input param as it's related to time not easy to verify, otherwise have to make clock provider as a dependency in many places.

Reply:
Don't you still need to have something like `EasyMock.expectLastCall().times(1)` to verify that the method has been called?

Reply:
From what I understand about EasyMock, the statement here is what to expect the call to happen, otherwise if a call to mocked object will result in the following error:

> java.lang.AssertionError: 
  Unexpected method call ServiceEmitter.emit(org.apache.druid.java.util.emitter.service.ServiceMetricEvent$Builder$1@e6516e):

Reply:
Hmm, need to check, merging the changes for now.